### DISCUSSION TITLE
SocialiteProvider - Apple

### DISCUSSION THREAD

**Discussion by mikeXdev** [SocialiteProviders/Providers#1145] [Q&A]
Hoping to get all your inputs with my problem.

We have two projects, both are built in Laravel Framework
1. Portal Website ( note, we have 2 portal site who is using SSO )
2. SSO 

In Portal Website, we have a button for Apple Sign In. It triggers to go to SSO Website with the parameters of [redirect_url, category, device type]. As soon as the redirect is triggered, I'm saving the parameters in the Session. After saving, executing the code to open the Sign Page. `Socialite::driver('apple')->stateless()->redirect()`.

After the apple sign in, it goes to the /callback route. Problem is, Sessions is now gone. I have no way of knowing where it should go back. 

Any idea how can I have pass a parameter to apple and retrieve the data in callback controller.

I also cannot do the response_mode `query` because I need to retrieve users information (email, first_name and last_name)

**Comment by hettiger** [comment]
Heres a workaround that also explains what's going on:

https://laracasts.com/discuss/channels/laravel/apple-login-with-socialite-clears-my-cart-session?page=1&replyId=930481
